test/Tests/src/Data/Time/Date_Part_Spec.enso

```diff
@@ -0,0 +1,99 @@
+            create_new_date 2000 3 31 . length_of_month . should_equal 31
+            create_new_date 2003 4 12 . length_of_month . should_equal 30
+            create_new_date 2017 5 16 . length_of_month . should_equal 31
+            create_new_date 2019 6 30 . length_of_month . should_equal 30
+            create_new_date 2019 7 5 . length_of_month . should_equal 31
+            create_new_date 2019 8 28 . length_of_month . should_equal 31
+            create_new_date 2019 9 30 . length_of_month . should_equal 30
+            create_new_date 2019 10 22 . length_of_month . should_equal 31
+            create_new_date 2019 11 7 . length_of_month . should_equal 30
+            create_new_date 2019 12 31 . length_of_month . should_equal 31
+
+        Test.specify "should return the day_of_year" <|
+            create_new_date 1990 9 18 . day_of_year . should_equal 261
+            create_new_date 1990 10 13 . day_of_year . should_equal 286
+            create_new_date 1992 9 28 . day_of_year . should_equal 272
+            create_new_date 1993 9 10 . day_of_year . should_equal 253
+            create_new_date 1996 3 15 . day_of_year . should_equal 75
+            create_new_date 2002 6 15 . day_of_year . should_equal 166
+            create_new_date 2007 5 4 . day_of_year . should_equal 124
+            create_new_date 2020 1 9 . day_of_year . should_equal 9
+            create_new_date 2020 4 9 . day_of_year . should_equal 100
+            create_new_date 2021 11 25 . day_of_year . should_equal 329
+            create_new_date 2023 3 21 . day_of_year . should_equal 80
+            create_new_date 2024 1 13 . day_of_year . should_equal 13
+
+        Test.specify "should return the day_of_week" <|
+            create_new_date 1990 9 18 . day_of_week . should_equal Day_Of_Week.Tuesday
+            create_new_date 1990 10 13 . day_of_week . should_equal Day_Of_Week.Saturday
+            create_new_date 1992 9 28 . day_of_week . should_equal Day_Of_Week.Monday
+            create_new_date 1993 9 10 . day_of_week . should_equal Day_Of_Week.Friday
+            create_new_date 1996 3 15 . day_of_week . should_equal Day_Of_Week.Friday
+            create_new_date 2002 6 15 . day_of_week . should_equal Day_Of_Week.Saturday
+            create_new_date 2007 5 4 . day_of_week . should_equal Day_Of_Week.Friday
+            create_new_date 2020 1 9 . day_of_week . should_equal Day_Of_Week.Thursday
+            create_new_date 2020 4 9 . day_of_week . should_equal Day_Of_Week.Thursday
+            create_new_date 2021 11 25 . day_of_week . should_equal Day_Of_Week.Thursday
+            create_new_date 2023 3 21 . day_of_week . should_equal Day_Of_Week.Tuesday
+            create_new_date 2024 1 13 . day_of_week . should_equal Day_Of_Week.Saturday
+
+        Test.specify "should return the correct week of year" <|
+            create_new_date 2021 8 1 . week_of_year Locale.mexico . should_equal 32
```

Comment:
Is there any significance of `Locale.mexico` here? If yes, can we get a comment? If not, I guess a comment saying its just a random locale will also help.